shoop/core/order_creator/_source.py

```diff
@@ -392,6 +395,19 @@ def get_product_lines(self):
         product_lines = [l for l in self.get_lines() if l.product]
         return product_lines
 
+    def get_suppliers(self):
+        return set([l.supplier for l in self.get_lines() if l.supplier])
+
+    def get_products_and_quantities(self, supplier=None):
+        q_counter = Counter()
+        for line in self.get_lines():
+            if supplier and line.supplier != supplier:
+                continue
+            product = line.product
+            q_counter[product] += line.quantity
+
+        return dict(q_counter)
+
```

Comment:
`get_supplier` and `get_products_and_quantities`: Make these private and move them under the `get_validation_errors` since they are only used there for now and IMHO we shouldn't make them public before there is any real use for them. (I don't see they would be very useful for OrderSource API users.)